File: src/JavaPackage/Debug.java
package JavaPackage;

import java.util.Arrays;

/**
 * Created by epicm on 08.12.2017.
 */
class Debug {

    static void fields(Object o){
        Arrays.stream(o.getClass().getDeclaredFields()).map(v -> {
            v.setAccessible(true);
            try {
                return "Pole: " + v.getName()
                        + " => " + v.getType() + ", " + v.get(o).toString();
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            }
            return "";
        }).forEach(System.out::println);
    }
}

File: src/ScalaPackage/Debug.scala
package ScalaPackage

/**
  * Created by epicm on 08.12.2017.
  */
trait Debug {
  def debugName(): Unit = println("Klasa: " + getClass.getSimpleName)
  def debugVars(): Unit = {
    getClass.getDeclaredFields.foreach( v => List.apply(v).map(v => {
      v.setAccessible(true)
      "Pole: " + v.getName + " => " + v.getType + ", " + v.get(this)
    }).foreach(s => println(s)))
  }
}
